php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1193 @@
+            CLASS_VARIABLE_DECLARATION(),
+            CLASS_CONSTANT_DECLARATION(),
+            TRAIT_USE_STATEMENT()
+        ));
+  }
+
+  public ClassPropertyDeclarationTree CLASS_CONSTANT_DECLARATION() {
+    return b.<ClassPropertyDeclarationTree>nonterminal(PHPLexicalGrammar.CLASS_CONSTANT_DECLARATION).is(
+      f.classConstantDeclaration(
+        b.token(PHPKeyword.CONST),
+        MEMBER_CONST_DECLARATION(),
+        b.zeroOrMore(f.newTuple97(b.token(COMMA), MEMBER_CONST_DECLARATION())),
+        EOS()));
+  }
+
+  public ConstantDeclarationTree CONSTANT_DECLARATION() {
+    return b.<ConstantDeclarationTree>nonterminal(PHPLexicalGrammar.CONSTANT_DECLARATION).is(
+      f.constantDeclaration(
+        b.token(PHPKeyword.CONST),
+        CONST_VAR(),
+        b.zeroOrMore(f.newTuple28(b.token(COMMA), CONST_VAR())),
+        EOS()));
+  }
+
+  public ClassPropertyDeclarationTree CLASS_VARIABLE_DECLARATION() {
+    return b.<ClassPropertyDeclarationTree>nonterminal(PHPLexicalGrammar.CLASS_VARIABLE_DECLARATION).is(
+      f.classVariableDeclaration(
+        b.firstOf(
+          f.singleToken(b.token(PHPKeyword.VAR)),
+          b.oneOrMore(MEMBER_MODIFIER())),
+        VARIABLE_DECLARATION(),
+        b.zeroOrMore(f.newTuple95(b.token(COMMA), VARIABLE_DECLARATION())),
+        EOS()));
+  }
+
+  public SyntaxToken MEMBER_MODIFIER() {
+    return b.<SyntaxToken>nonterminal().is(
+      b.firstOf(
+        b.token(PHPKeyword.PUBLIC),
+        b.token(PHPKeyword.PROTECTED),
+        b.token(PHPKeyword.PRIVATE),
+        b.token(PHPKeyword.STATIC),
+        b.token(PHPKeyword.ABSTRACT),
+        b.token(PHPKeyword.FINAL)
+      ));
+  }
+
+  public MethodDeclarationTree METHOD_DECLARATION() {
+    return b.<MethodDeclarationTree>nonterminal(PHPLexicalGrammar.METHOD_DECLARATION).is(
+      f.methodDeclaration(
+        b.zeroOrMore(MEMBER_MODIFIER()),
+        b.token(PHPKeyword.FUNCTION),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        IDENTIFIER(),
+        PARAMETER_LIST(),
+        b.firstOf(
+          EOS(),
+          BLOCK())
+      ));
+  }
+
+  public FunctionDeclarationTree FUNCTION_DECLARATION() {
+    return b.<FunctionDeclarationTree>nonterminal(PHPLexicalGrammar.FUNCTION_DECLARATION).is(
+      f.functionDeclaration(
+        b.token(PHPKeyword.FUNCTION),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        IDENTIFIER(),
+        PARAMETER_LIST(),
+        BLOCK()));
+  }
+
+  public ParameterListTree PARAMETER_LIST() {
+    return b.<ParameterListTree>nonterminal(PHPLexicalGrammar.PARAMETER_LIST).is(
+      f.parameterList(
+        b.token(PHPPunctuator.LPARENTHESIS),
+        b.optional(
+          f.newTuple94(
+            PARAMETER(),
+            b.zeroOrMore(
+              f.newTuple93(
+                b.token(PHPPunctuator.COMMA),
+                PARAMETER())))),
+        b.token(PHPPunctuator.RPARENTHESIS)));
+  }
+
+  public ParameterTree PARAMETER() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)